core-bundle/src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -971,6 +971,77 @@ public static function ampersand($strString, $blnEncode=true): string
 	{
 		return preg_replace('/&(amp;)?/i', ($blnEncode ? '&amp;' : '&'), $strString);
 	}
+
+	/**
+	 * Converts an input-encoded string back to its raw UTF-8 value it originated from.
+	 *
+	 * It handles all Contao input encoding specifics like insert tags, basic entities and encoded entities.
+	 *
+	 * @param bool $blnRemoveInsertTags True to remove insert tags instead of replacing them
+	 */
+	public static function getRawDecodedValue(string $strValue, bool $blnRemoveInsertTags = false): string
+	{
+		if ($blnRemoveInsertTags)
+		{
+			$strValue = static::stripInsertTags($strValue);
+		}
+		else
+		{
+			$strValue = Controller::replaceInsertTags($strValue, false);
+		}
+
+		$strValue = strip_tags($strValue);
+		$strValue = static::restoreBasicEntities($strValue);
+		$strValue = static::decodeEntities($strValue);
+
+		// Ensure valid UTF-8
+		if (preg_match('//u', $strValue) !== 1)
+		{
+			$substituteCharacter = mb_substitute_character();
+			mb_substitute_character(0xFFFD);
+
+			$strValue = mb_convert_encoding($strValue, 'UTF-8', 'UTF-8');
```

Comment:
this means we no longer support anything else than UTF-8 right? I'm totally fine with that, but isn't there a localconfig for the character encoding?

Reply:
I think we should deprecate the character encoding in the config and force `UTF-8`. See also #3015

Reply:
deprecate means we still need to support it though? Or we agree that this is an acceptable break in the next minor 🙃

Reply:
Honestly, nothing ever worked in anything other than UTF-8 because why would it? Imho this config can be removed anywhere. I don't even understand why anyone would have ever used that.

Reply:
Back in 2006, not everyone had UTF-8 or was using UTF-8, therefore the charset was configurable (and still is). While I agree that we could enforce UTF-8, it is still an unnecessary BC break.

Reply:
I can add a check if the config charset is `utf-8` and only then execute this part of the code.

Reply:
So, what do we do here? I would leave it as it is in this PR and add the deprecation for the config charset in #3015

Reply:
Two options for me:
 1. can we just use the `Config::get('characterSet')` as argument for `mb_convert_encoding`?
 2. Alternatively, let's create a new issue assigned to 4.12 to remove encoding (or check on this again)

Reply:
Also, did you see there's a `StringUtil::convertEncoding()` method that does pretty much the same?

Reply:
> Alternatively, let's create a new issue assigned to 4.12 to remove encoding (or check on this again)

Done https://github.com/contao/contao/issues/3050